### PR TITLE
[LoongArch64] Optimize the overflow check for `GT_ADD` and `GT_SUB` in emitter::emitInsTernary().

### PR DESCRIPTION
**Pull request by LuckyXu-HF**
This PR is part of the issue https://github.com/dotnet/runtime/issues/69705 to amend the LA's port.

Optimize the overflow check for `GT_ADD` and `GT_SUB` in emitter::emitInsTernary().

### CODE REVIEW

**Review by shushanhf — APPROVED** [review]
Thanks!
LGTM

**Comment by LuckyXu-HF** [comment]
hi @jakobbotsch , Could you please review this PR? Thank you very much.

File: src/coreclr/jit/emitloongarch64.cpp [hunk 1]
```diff
@@ -5048,10 +5048,14 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
 
         // TODO-LOONGARCH64-CQ: here sign-extend dst when deal with 32bit data is too conservative.
         if (EA_SIZE(attr) == EA_4BYTE)
+        {
             emitIns_R_R_I(INS_slli_w, attr, dst->GetRegNum(), dst->GetRegNum(), 0);
+        }
     }
     else
     {
+        assert((dst->OperGet() == GT_ADD) || (dst->OperGet() == GT_SUB));
```
Comment (jakobbotsch) [inline src/coreclr/jit/emitloongarch64.cpp:5057]:
```suggestion
        assert(dst->OperIs(GT_ADD, GT_SUB));
```

similarly elsewhere.

File: src/coreclr/jit/emitloongarch64.cpp [hunk 2]
```diff
@@ -5064,113 +5068,94 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
...
+                    emitIns_R_R_R(INS_or, attr, REG_R21, regOp1, REG_R0);
+                }
+                else
+                {
+                    saveOperReg1 = regOp1;
+                }
             }
-            else
+
+            if (!(dst->gtFlags & GTF_UNSIGNED))
```
Comment (jakobbotsch) [inline src/coreclr/jit/emitloongarch64.cpp:5090]:
```suggestion
            if ((dst->gtFlags & GTF_UNSIGNED) == 0)
```
(Coding conventions)

**Review by jakobbotsch — APPROVED** [review]
LGTM. Feel free to include the style nits in future PRs.

**Comment by jakobbotsch** [comment]
Looks like there are jit-format failures -- can you please fix those?

**Comment by LuckyXu-HF** [comment]
> LGTM. Feel free to include the style nits in future PRs.

Thank you very much for pointing out these! I will pay extra attention in future PRs.

**Comment by LuckyXu-HF** [comment]
> Looks like there are jit-format failures -- can you please fix those?

I'm sorry, did you mean this https://github.com/dotnet/runtime/pull/98742#issuecomment-1956337062 jit-format failures?

`libraries.pmi.linux.arm64.checked.mch	165,051,190	165,061,123	+0.01%`

It seems to be an increase of instructions for arm64, but I think this commit should not affect arm64.
I will try to fix it, thank you.

**Comment by jakobbotsch** [comment]
> > Looks like there are jit-format failures -- can you please fix those?
> 
> I'm sorry, did you mean this [#98742 (comment)](https://github.com/dotnet/runtime/pull/98742#issuecomment-1956337062) jit-format failures?
> 
> `libraries.pmi.linux.arm64.checked.mch 165,051,190 165,061,123 +0.01%`
> 
> It seems to be an increase of instructions for arm64, but I think this commit should not affect arm64. I will try to fix it, thank you.

No, I mean the failures in the "runtime (Build Formatting linux x64)" and "runtime (Build Formatting windows x64)" CI checks. 
https://dev.azure.com/dnceng-public/public/_build/results?buildId=580692&view=logs&jobId=16595cbc-0780-584c-1db4-eb479b5d69f7&j=16595cbc-0780-584c-1db4-eb479b5d69f7&t=02d65da2-975f-54f5-198e-6c8a545d5505 has information for how to fix the formatting errors.

**Comment by LuckyXu-HF** [comment]
> > > Looks like there are jit-format failures -- can you please fix those?
> > 
> > 
> > I'm sorry, did you mean this [#98742 (comment)](https://github.com/dotnet/runtime/pull/98742#issuecomment-1956337062) jit-format failures?
> > `libraries.pmi.linux.arm64.checked.mch 165,051,190 165,061,123 +0.01%`
> > It seems to be an increase of instructions for arm64, but I think this commit should not affect arm64. I will try to fix it, thank you.
> 
> No, I mean the failures in the "runtime (Build Formatting linux x64)" and "runtime (Build Formatting windows x64)" CI checks. https://dev.azure.com/dnceng-public/public/_build/results?buildId=580692&view=logs&jobId=16595cbc-0780-584c-1db4-eb479b5d69f7&j=16595cbc-0780-584c-1db4-eb479b5d69f7&t=02d65da2-975f-54f5-198e-6c8a545d5505 has information for how to fix the formatting errors.

Oh, of course! The CI check prompts below, I missed this prompt.
I will have a look and try to fix it. Thank you for your guidance and suggestions!

File: src/coreclr/jit/emitloongarch64.cpp (hunk 1 above)
Comment (LuckyXu-HF) [reply src/coreclr/jit/emitloongarch64.cpp]: Already updated here and elsewhere in this commit, thank you very much!

File: src/coreclr/jit/emitloongarch64.cpp (hunk 2 above)
Comment (LuckyXu-HF) [reply src/coreclr/jit/emitloongarch64.cpp]: Thank you! I will pay attention to these in future PRs.

**Comment by LuckyXu-HF** [comment]
Hi @jakobbotsch , I have fixed the jit-format failures. The remaining check failures looks like unrelated to this commit.
Could you please take a look? Thanks.

**Pull request merged**